yarn-project/noir-contracts/src/contracts/account_contract/src/entrypoint.nr

```diff
@@ -52,4 +53,73 @@ impl EntrypointPayload {
     fields = fields.push(self.nonce);
     fields.storage
   }
+
+  // Serializes the payload as an array of bytes. Useful for hashing with sha256.
+  fn to_be_bytes(self) -> [u8; ENTRYPOINT_PAYLOAD_SIZE_IN_BYTES] {
+    let mut bytes: [u8; ENTRYPOINT_PAYLOAD_SIZE_IN_BYTES] = [0; ENTRYPOINT_PAYLOAD_SIZE_IN_BYTES];
+
+    let args_len = self.flattened_args.len();
+    let selectors_len = self.flattened_selectors.len();
+    let targets_len = self.flattened_targets.len();
+
+    for i in 0..args_len {
+      let item_bytes = self.flattened_args[i].to_be_bytes(32);
+      for j in 0..32 { 
+        bytes[i * 32 + j] = item_bytes[j];
+      }
+    }
+
+    for i in 0..selectors_len {
+      let item_bytes = self.flattened_selectors[i].to_be_bytes(32);
+      for j in 0..32 { 
+        bytes[args_len * 32 + i * 32 + j] = item_bytes[j];
+      }
+    }
+
+    for i in 0..targets_len {
+      let item_bytes = self.flattened_targets[i].to_be_bytes(32);
+      for j in 0..32 { 
+        bytes[(args_len + selectors_len) * 32 + i * 32 + j] = item_bytes[j];
+      }
+    }
+    
+    let item_bytes = self.nonce.to_be_bytes(32);
+    for j in 0..32 { 
+      bytes[(args_len + selectors_len + targets_len) * 32 + j] = item_bytes[j];
+    }
+
+    bytes
+  }
+
+  // TODO: This function returns all zeroes. Figure out why!
```

Comment:
@sirasistant this function returned a byte array with just zeroes. Any idea why? I spent a few hours debugging and eventually gave up and switched to the implementation above.